R/calc_weights_and_outputs.R

```diff
@@ -178,3 +185,44 @@ calc_weights_and_outputs <- function(
   }
 
 }
+
+calc_weights_prechecks <- function(
+  total_portfolio,
+  portfolio_type,
+  output_dir,
+  data_dir
+) {
+  if (is.null(total_portfolio)) {
+    log_warn("Portfolio is NULL.")
+  } else if (nrow(total_portfolio) == 0L) {
+    log_error("Portfolio has no rows.")
+    stop("Portfolio has no rows.")
+  }
```

Comment:
I'm passing in `NULL` during the prechecks for `run_pacta()` (see `R/run_pacta.R`, line 69). at that point, the `total_portfolio` file isn't generated yet, so this is a workaround.

I'd be open to other options (maybe a check_portfolio flag?) if you think that's better

Reply:
Ah, yeah I see the point.

I guess the flag should really indicate something along the lines of "should I expect output files from the audit process to have been created already?" 

So maybe something along the lines of:
``` r
  analysis_prechecks(
    total_portfolio_path = NULL,
    pacta_data_dir = pacta_data_dir,
    output_dir = output_dir,
    check_audit_outputs = TRUE
  )
```

and in `run_pacta` it would be: 
``` r
  analysis_prechecks(
   ...,
    check_audit_outputs = FALSE
  )
```

Reply:
I'm always hesitant to use `NULL` in a function's default/ expected behaviour...

Reply:
https://github.com/RMI-PACTA/workflow.pacta/issues/85